### PR TITLE
added Dao and Services for Order, ShoppingCart and User

### PR DESCRIPTION
**Pull request by TimoxaV**

### CODE REVIEW

File: src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order(shoppingCart.getUserId());
+        order.setProducts(shoppingCart.getProducts());
```
Comment (MaliukDaria) [inline src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java]: from checklist: In completeOrder() use List.copyOf(shoppingCart.getProducts()) to copy products from shoppingCart to order and don’t forget to clear ShoppingCart after you complete user’s order.

File: src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        orderDao.create(order);
+        return order;
```
Comment (MaliukDaria) [inline src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java]:
```suggestion
        return orderDao.create(order);;
```

File: src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        return orderDao.getAllOrders().stream()
+                .filter(order -> order.getUserId().equals(userId))
+                .collect(Collectors.toList());
```
Comment (MaliukDaria) [inline src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java]: from checklist: In “read” operations filter data only on Dao layer, so on Service layer you receive “ready to use” data.

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        shoppingCart.setProducts(cartProducts);
+        return shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts()
+                .removeIf(productSearch -> productSearch.getId().equals(product.getId()));
+        shoppingCartDao.update(shoppingCart);
+        return true;
```
Comment (MaliukDaria) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: если такого продукта нету в корзине, все равно будет всегда возвращаться true?

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        shoppingCart.setProducts(new ArrayList<>());
+        shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getByUserId(Long userId) {
+        return shoppingCartDao.getAllCarts().stream()
+                .filter(shoppingCart -> shoppingCart.getUserId().equals(userId))
+                .findFirst()
+                .get();
```
Comment (MaliukDaria) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: checklist: In “read” operations filter data only on Dao layer, so on Service layer you receive “ready to use” data.

File: src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java (hunk 1 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java]: Thanks. Fixed.

File: src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java (hunk 2 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java]: done

File: src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java (hunk 3 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java]: fixed

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 4 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: fixed

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 5 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: fixed

File: src/main/java/com/internet/shop/dao/OrderDao.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import com.internet.shop.model.Order;
+import java.util.List;
+import java.util.Optional;
+
+public interface OrderDao {
+    Order create(Order order);
+
+    Optional<Order> getById(Long orderId);
+
+    List<Order> getByUserId(Long userId);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/OrderDao.java]:
```suggestion
    List<Order> getUserOrders(Long userId);
```

File: src/main/java/com/internet/shop/dao/OrderDao.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+    Order update(Order order);
+
+    boolean deleteById(Long orderId);
+
+    List<Order> getAllOrders();
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/OrderDao.java]:
```suggestion
    List<Order> getAll();
```

File: src/main/java/com/internet/shop/dao/ShoppingCartDao.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+    Optional<ShoppingCart> getById(Long shoppingCartId);
+
+    Optional<ShoppingCart> getByUserId(Long userId);
+
+    ShoppingCart update(ShoppingCart shoppingCart);
+
+    boolean deleteById(Long shoppingCartId);
+
+    List<ShoppingCart> getAllCarts();
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/ShoppingCartDao.java]:
```suggestion
    List<ShoppingCart> getAll();
```

File: src/main/java/com/internet/shop/db/Storage.java [hunk 9]
```diff
@@ -1,16 +1,43 @@
...
 
     public static void addProduct(Product product) {
         productId++;
         product.setId(productId);
         products.add(product);
     }
+
+    public static void addUser(User user) {
+        userId++;
+        user.setId(userId);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/db/Storage.java]:
```suggestion
        user.setId(++userId);
```
same in other methods

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        return shoppingCartDao.create(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        List<Product> cartProducts = shoppingCart.getProducts();
+        cartProducts.add(product);
+        shoppingCart.setProducts(cartProducts);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
        shoppingCart.getProducts().add(product);
```

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+        return shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        if (shoppingCart.getProducts()
+                .removeIf(productSearch -> productSearch.getId().equals(product.getId()))) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
        if (shoppingCart.getProducts().remove(product)) {
```

File: src/main/java/com/internet/shop/dao/OrderDao.java (hunk 6 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/dao/OrderDao.java]: done

File: src/main/java/com/internet/shop/dao/OrderDao.java (hunk 7 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/dao/OrderDao.java]: done

File: src/main/java/com/internet/shop/dao/ShoppingCartDao.java (hunk 8 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/dao/ShoppingCartDao.java]: done

File: src/main/java/com/internet/shop/db/Storage.java (hunk 9 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/db/Storage.java]: done

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 10 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: done

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 11 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**